invoker/conformance/src/main/java/com/google/cloud/functions/conformance/BackgroundEventConformanceFunction.java

```diff
@@ -0,0 +1,59 @@
+package com.google.cloud.functions.conformance;
+
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.cloud.functions.Context;
+import com.google.cloud.functions.RawBackgroundFunction;
+import java.io.BufferedWriter;
+import java.io.FileWriter;
+
+/**
+ * This class is used by the Functions Framework Conformance Tools to validate
+ * the framework's Background Event API. It can be run with the following
+ * command:
+ *
+ * <pre>
+ * {@code
+ * $ functions-framework-conformance-client \
+ *   -cmd="mvn function:run -Drun.functionTarget=com.google.cloud.functions.conformance.BackgroundEventConformanceFunction" \
+ *   -type=legacyevent \
+ *   -buildpacks=false \
+ *   -validate-mapping=false \
+ *   -start-delay=10
+ * }
+ * </pre>
+ */
+public class BackgroundEventConformanceFunction implements RawBackgroundFunction {
+
+  private static final Gson gson = new GsonBuilder().serializeNulls().setPrettyPrinting().create();
+
+  @Override
+  public void accept(String data, Context context) throws Exception {
+    try (BufferedWriter writer = new BufferedWriter(new FileWriter("function_output.json"))) {
+      writer.write(serialize(data, context));
+    }
+  }
+
+  /**
+   * Create a structured JSON representation of the request context and data
+   */
+  private String serialize(String data, Context context) {
+    JsonObject contextJson = new JsonObject();
+    contextJson.addProperty("eventId", context.eventId());
+    contextJson.addProperty("timestamp", context.timestamp());
+    contextJson.addProperty("eventType", context.eventType());
+    
+    JsonElement resource = gson.fromJson(context.resource(), JsonElement.class);
+    contextJson.add("resource", resource);
+
+    JsonObject dataJson = gson.fromJson(data, JsonObject.class);
+
+    JsonObject json = new JsonObject();
+    json.add("data", dataJson);
+    json.add("context", contextJson);
+    return gson.toJson(json);
+  }
```

Comment:
I feel like this method should not be in the conformance test repo, but rather in the `Event.java` file:

https://github.com/GoogleCloudPlatform/functions-framework-java/blob/master/invoker/core/src/main/java/com/google/cloud/functions/invoker/Event.java

Or something similar.

---

This implementation also seems a bit manual for serialization.

Reply:
this is analogous to what we do in other conformance tests eg: https://github.com/GoogleCloudPlatform/functions-framework-nodejs/blob/master/test/conformance/function.js#L15-L24

Compared to Javascript, there is a bit of extra logic needed here because the "data" argument gets passed as a JSON string so we need deserialize before we re-serialize in order to prevent it from being written as an escaped string.